lib/rspec/support/ruby_features.rb

```diff
@@ -101,7 +105,7 @@ def supports_taint?
       end
       ripper_requirements = [ComparableVersion.new(RUBY_VERSION) >= '1.9.2']
 
-      ripper_requirements.push(false) if Ruby.rbx?
+      ripper_requirements.push(false) if Ruby.rbx? || Ruby.truffleruby?
```

Comment:
This is not ideal, because TruffleRuby might support `Ripper` at some point.
How about checking `Ripper.respond_to?(:lex)` which is what https://github.com/k0kubun/hamlit does for instance?

Reply:
You could push that onto the stack for if run truffle ruby?

Reply:
@eregon I think there would be an issue with this because part of the spec asserts `Ripper` would not be loaded by feature detection: https://github.com/rspec/rspec-support/blob/b83b4cecb3bb56300addae89cb0c437f1af906fb/spec/rspec/support/ruby_features_spec.rb#L151-L153

Reply:
I see, that is unfortunate. Probably fine to keep as-is then and change it again (e.g., to a version check) once TruffleRuby supports Ripper.

Reply:
```suggestion
      ripper_requirements.push(defined?(::Ripper) && Ripper.respond_to?(:lex)) if Ruby.truffleruby?
```

Reply:
That would not work because to access the Ripper constant one would need to `require 'ripper'` but there is a spec checking that as @bjfish wrote above.

Reply:
Although that spec would likely not see whether `ripper` was required before `ripper_supported?` is defined. But I guess the intent is that file doesn't load ripper.

Reply:
@eregon `defined?(::Ripper)` would return false unless you've required it, which is the same as this behaviour, but then if it has been required, it will check for the `lex` method to see if truffle ruby is now compatible with ripper support.

Reply:
How does this work in practice in RSpec? If it wants to use Ripper, it will have to require it, no?
Or does it rely on the user requiring Ripper to use it?

Reply:
> How does this work in practice in RSpec? If RSpec wants to use Ripper, it will have to require it, no? Or does it rely on the user requiring Ripper to use it?

We require it if ripper is supported, but on recent MRI it's already loaded. In practise it would mean that you'd have to have require it on truffle ruby if you wanted to use it and it would work.

Reply:
> We require it if ripper is supported,

Right, so the whole point of this logic is then avoiding `require 'ripper'` if Ripper is unsupported.
Is that to avoid changing `$LOADED_FEATURES` or making sure code which uses Ripper still `require 'ripper'`, but only for Rubies with unsupported Ripper?

I would suggest something a lot more straightforward by always requiring Ripper, then we wouldn't need to have so much Ruby-implementation-specific code here.

> on recent MRI it's already loaded

I don't see that:
```
$ ruby -ve 'puts $".grep(/ripper/i)'
ruby 2.7.0p0 (2019-12-25 revision 647ee6f091) [x86_64-linux]
```

> In practise it would mean that you'd have to have require it on truffle ruby if you wanted to use it and it would work.

I wouldn't want that, I'd want the same behavior as MRI.
Since anyway this does not seem a critical feature of RSpec, I propose to adapt the code here once TruffleRuby supports Ripper and merge the PR as it is.

Reply:
>Right, so the whole point of this logic is then avoiding require 'ripper' if Ripper is unsupported. Is that to avoid changing $LOADED_FEATURES or making sure code which uses Ripper still require 'ripper', but only for Rubies with unsupported Ripper?

It's to avoid requiring ripper if possible.

> I would suggest something a lot more straightforward by always requiring Ripper, then we wouldn't need to have so much Ruby-implementation-specific code here.

We do the least requiring we can to avoid false positives due to poisoned environments.

> I don't see that:

I stand corrected, `irb` requires it, `ruby` does not.

> I wouldn't want that, I'd want the same behavior as MRI.

In my experience only MRI behaves like MRI, other Ruby engines always have quirks. 😢

Reply:
> In my experience only MRI behaves like MRI, other Ruby engines always have quirks.

FWIW, TruffleRuby tries hard to behave exactly like MRI. But implementing Ripper is a significant effort, so that's not done yet.

Reply:
To be fair I should probably say that every engine has quirks, it's just the MRI ones are the reference quirks? 🤷‍♂

Reply:
I just wanted to end on a positive note :)
This PR is not the most beautiful code but TruffleRuby otherwise runs RSpec (and Rails nowadays) unmodified with no TruffleRuby-specific code since many years now.
We try hard to avoid any TruffleRuby-specific code in gems.